astropy/cosmology/_utils.py

```diff
@@ -21,7 +21,9 @@
 if TYPE_CHECKING:
     from astropy.cosmology import Parameter
 
+
 _F = TypeVar("_F", bound=Callable[..., Any])
+R = TypeVar("R")
```

Comment:
Annoyingly, `ruff` can track whether private TypeVars are needed (or not) but it doesn't track public ones!
I think this is because not every package uses `__all__`, so it's not something `ruff` can rely on. It would be great if there were a setting saying it could use `__all__` to assess what's exported.

Reply:
Addressed. Thanks!